src/wallet.js

```diff
@@ -195,13 +193,34 @@ MyWallet.makePairingCode = function (success, error) {
   }
 };
 
-MyWallet.loginFromJSON = function (stringWallet, stringExternal, magicHashHex, password) {
+MyWallet.loginFromJSON = function (stringWallet, stringExternal, magicHashHexExternal, stringLabels, magicHashHexLabels, password) {
+  assert(stringWallet, 'Wallet JSON required');
+
+  // If metadata service returned 404, do not pass in a string.
+  var externalJSON = null;
+  var labelsJSON = null;
+
+  if (stringExternal) {
+    assert(magicHashHexExternal, 'Magic hash for external required');
```

Comment:
Given the complexity in bridging iOs with Javascript, I think it's important to make sure `magicHash` is never `undefined` by accident.

We decided early on to only create a buy-sell KV-store entry for wallets that use this feature. This was enforced by the fact that the wallet simply doesn't try to save this entry before an exchange account is created.

I changed the `toJSON()` method to return `undefined` if no exchange account exists. This is distinct from `null`: `JSON.toString(null) => 'null'` vs.. JSON.toString(undefined) => undefined. The `save()` won't do anything in this case either.

Also note that `stringLabels` and `magicHashHexLabels` are required for buy-sell to work correctly, but they are not required for `loginFromJSON()` since there might be other reasons to use it than buy-sell.